batch7crud-roll3/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,181 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.lftechnology.batch7crud.entity.Student;
+import com.lftechnology.batch7crud.exception.DataException;
+import com.lftechnology.batch7crud.service.StudentService;
+
+/**
+ * Servlet implementation class StudentController
+ */
+@WebServlet("/student/*")
+public class StudentController extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+		// TODO Auto-generated method stub
+		String pathInfo = request.getPathInfo();
+		if (pathInfo == null) {
+			list(request, response, 1);
+		} else {
+			String[] pathParts = pathInfo.split("/");
+
+			if (pathParts[1].equals("create")) {
+				create(request, response);
+			} else if (pathParts[1] != null && pathParts.length == 2) {
+				int page = 1;
+				if (pathParts[1] != null) {
+					page = Integer.parseInt(pathParts[1]);
+				}
+				list(request, response, page);
+			} else if (pathParts[1] != null && pathParts[2].equals("edit")) {
+				edit(request, response, Integer.parseInt(pathParts[1]));
+
+			} else if (pathParts[1] != null && pathParts[2].equals("delete")) {
+				deleteProcess(request, response, Integer.parseInt(pathParts[1]));
+
+			} else {
+				response.sendRedirect("/batch7crud-roll3/student");
+			}
+		}
+	}
+
+	protected void doPost(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+		String pathInfo = request.getPathInfo();
+		String[] pathParts = pathInfo.split("/");
+		
+		try {
+			if (pathParts[1].equals("create")) {
+				createProcess(request, response);
+			} else if (pathParts[2].equals("edit")) {
+				editProcess(request, response, Integer.parseInt(pathParts[1]));
+			} else if (pathParts[2].equals("delete")) {
+				deleteProcess(request, response, Integer.parseInt(pathParts[1]));
+			} else {
+				list(request, response, 1);
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	private void list(HttpServletRequest request, HttpServletResponse response, int page)
+			throws ServletException, IOException {
+		try {
+			int pageSize = 3;
+			StudentService stdService = new StudentService();
+			List<Student> stdList = stdService.fetch(page, pageSize);
+			int count = stdService.fetchTotal();
+			
+			request.setAttribute("stdList", stdList);
+			request.setAttribute("pageSize", pageSize);
+
+			request.setAttribute("count", count);
+			request.setAttribute("pageNum", page);
+			request.getRequestDispatcher("/WEB-INF/views/student/list.jsp").forward(request, response);
+		} catch (DataException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			request.setAttribute("message", e.getMessage());
+			request.getRequestDispatcher("/WEB-INF/views/error.jsp").forward(request, response);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/WEB-INF/views/error.jsp" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)